docs/developer-guide/integrations/users/openId.md

```diff
@@ -0,0 +1,168 @@
+
+- Open Google developer console and, from credentials section, create a new credential of type **Oauth client ID**
+
+<img src="../img/google-create-credentials.jpg" class="ms-docimage"  style="max-width:500px;"/>
+
+- Set the **Application Type** to **Web Application**, name it as you prefer and configure the root of the application as an authorized redirect URI. Then click on **Create**
+
+<img src="../img/google-create-oauth2.jpg" class="ms-docimage"  style="max-width:500px;"/>
+
+- After creation you will obtain **ClientID** and **Client Secret** to use to configure MapStore.
+
+Please follow the [Google documentation](https://developers.google.com/identity/protocols/oauth2/openid-connect) for any detail or additional configuration.
+
+#### Configure MapStore back-end for Google OpenID
+
+After the setup, you will have to:
+
+- create/edit `mapstore-ovr.properties` file (in data-dir or class path) to configure the google provider this way:
+
+```properties
+
+# enables the google OpenID Connect filter
+googleOAuth2Config.enabled=true
+
+#clientId and clientSecret
+googleOAuth2Config.clientId=<the_client_id_from_google_dev_console>
+googleOAuth2Config.clientSecret=<the_client_secret_from_google_dev_console>
+
+# create the user if not present
+googleOAuth2Config.autoCreateUser=true
+
+# Redirect URL
+googleOAuth2Config.redirectUri=https://<your-appliction-domain>/mapstore/rest/geostore/openid/google/callback
+# Internal redirect URI (you can set it to relative path like this `../../..` to make this config work across domain)
+googleOAuth2Config.internalRedirectUri=https://<your-appliction-domain>/mapstore/
+
+## discoveryUrl: contains all the information for the specific service.
+googleOAuth2Config.discoveryUrl=https://accounts.google.com/.well-known/openid-configuration
+#If the `discoveryUrl` has not been specified, you can manually configure the following options.
+```
+
+#### Configure MapStore front-end for Google OpenID
+
+- Add an entry for `google` in `authenticationProviders` inside `localConfig.json` file.
+
+```json
+{
+    "authenticationProviders": [
+      {
+        "type": "openID",
+        "provider": "google"
+      },
+      {
+        "type": "basic",
+        "provider": "geostore"
+      }
+    ]
+}
+```
+
+### Keycloak
+
+[Keycloak](https://www.keycloak.org/) is an open source identity and access management application widely used. MapStore has the ability to integrate with keycloak:
+
+- Using the standard OpenID protocol
+- Supporting SSO (not yet implemented)
+- Integrating with users and roles, as well as for ldap. (not yet implemented)
+
+In this section you can see how to configure keycloak as a standard OpenID provider
+
+#### Configure keycloak Client
+
+Create a new Client on keycloak. In this guide we will name it `mapstore-server` (because if you need to configure SSO, we may need another key to call `mapstore-client`)
```

Comment:
"we may need another key to call `mapstore-client"

can you elaborate a bit more here?

Reply:
It will have a dedicated section in doc when inplemented.